Assets/Code/Scripts/Fight/FightCells/Impediments/TrapSO.cs

```diff
@@ -13,24 +17,74 @@ namespace FrostfallSaga.Fight.FightCells.Impediments
     [CreateAssetMenu(fileName = "Trap", menuName = "ScriptableObjects/Fight/Impediments/Trap", order = 0)]
     public class TrapSO : AImpedimentSO
     {
+        [field: SerializeField] public Targeter Targeter { get; private set; }
+        [field: SerializeField] public IImpedimentAnimationSO Animation { get; private set; }
         [field: SerializeField, Header("Trap definition")] public ETrapTriggerTime[] TriggerTimes { get; private set; }
-        [SerializeReference] public AEffect[] Effects = { };
+        [SerializeReference] public AEffect[] OtherCellsEffects = { };
+        [SerializeReference] public AEffect[] OnCellEffects = { };
+        [SerializeReference] public AFightCellAlteration[] CellAlterations = { };
+        [SerializeField] public ETrapType TrapType;
```

Comment:
Why do you need this trap type?

Reply:
In order to know if a trap of the same type ( fire, pitfall, explosion and spikes )  is already put on a cell

Reply:
Okay I get it!

You can just set the setter as private here
```suggestion
        [SerializeField] public ETrapType TrapType { get; private set; }
```